ecs-cli/modules/cli/compose/entity/service/service_test.go

```diff
@@ -767,6 +779,60 @@ func getUpdateServiceMockClient(t *testing.T,
 	return mockEcs
 }
 
+func getCreateServiceMockClient(t *testing.T,
+	ctrl *gomock.Controller,
+	taskDefinition ecs.TaskDefinition,
+	taskDefID string,
+	registerTaskDefResponse ecs.TaskDefinition,
+) *mock_ecs.MockECSClient {
+
+	mockEcs := mock_ecs.NewMockECSClient(ctrl)
+
+	createdService := &ecs.Service{
+		TaskDefinition: aws.String("arn/test-task-def"),
+		Status:         aws.String("ACTIVE"),
+		DesiredCount:   aws.Int64(0),
+		RunningCount:   aws.Int64(0),
+		ServiceName:    aws.String("test-created"),
+	}
+	updatedService := *createdService
+	gomock.InOrder(
+		mockEcs.EXPECT().DescribeService(gomock.Any()).Return(getDescribeServiceTestResponse(nil), nil),
+
+		mockEcs.EXPECT().RegisterTaskDefinitionIfNeeded(gomock.Any(), gomock.Any()).Do(func(x, y interface{}) {
+			verifyTaskDefinitionInput(t, taskDefinition, x.(*ecs.RegisterTaskDefinitionInput))
+		}).Return(&registerTaskDefResponse, nil),
+
+		mockEcs.EXPECT().CreateService(
+			gomock.Any(), // serviceName
+			gomock.Any(), // taskDefName
+			gomock.Any(), // loadBalancer
+			gomock.Any(), // role
+			gomock.Any(), // deploymentConfig
+			gomock.Any(), // networkConfig
+			gomock.Any(), // launchType
+			gomock.Any(), // healthCheckGracePeriod
+		).Do(func(a, b, c, d, e, f, g, h interface{}) {
+			observedTaskDefID := b.(string)
+			assert.Equal(t, taskDefID, observedTaskDefID, "Task Definition name should match")
+		}).Return(nil),
+
+		mockEcs.EXPECT().DescribeService(gomock.Any()).Return(getDescribeServiceTestResponse(nil), nil),
+		mockEcs.EXPECT().DescribeService(gomock.Any()).Return(getDescribeServiceTestResponse(createdService), nil).MaxTimes(2),
```

Comment:
these 2 lines (the two DescribeService calls - one returning nil, the 2nd service details) seem specific to testing the case where the service is not immediately describable. Can we make that apparent by renaming this func `getCreateServiceWithDelayMockClient` ?